src/StaticCaching/CustomInvalidator.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Rapidez\Statamic\StaticCaching;
+
+use Statamic\Eloquent\Globals\GlobalSet;
+use Statamic\Eloquent\Structures\Nav;
+use Statamic\Entries\Entry;
+use Statamic\StaticCaching\DefaultInvalidator;
+use Statamic\Support\Str;
+
+class CustomInvalidator extends DefaultInvalidator
+{
+    public function invalidate($item)
+    {
+        if ($item instanceof GlobalSet || $item instanceof Nav || $this->rules === 'all') {
+            return $this->cacher->flush();
+        }
+
+        $urls = [];
+
+        if ($item instanceof Entry && $item->collectionHandle() === 'categories' && $item->linked_category) {
+            $urls[] = Str::ensureLeft($item->linked_category['url_path'], '/');
+        }
+
+        if ($item instanceof Entry && $item->uri()) {
+            $urls[] = $item->uri();
+        }
```

Comment:
This will cover products and brands? Isn't this already covered by the default invalidator?